### PR TITLE
refactor(components/lookup): implement strict-mode for lookup, search, and country field

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**

### CODE REVIEW

File: libs/components/indicators/src/lib/modules/tokens/tokens.component.ts [hunk 1]
```diff
@@ -207,7 +207,7 @@ export class SkyTokensComponent implements OnDestroy {
   private _disabled: boolean;
   private _dismissible: boolean;
   private _focusable: boolean;
-  private _tokens: SkyToken[];
+  private _tokens: SkyToken[] | undefined;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/indicators/src/lib/modules/tokens/tokens.component.ts:210]: ☝️ I'll create a separate PR to refactor the Indicators stuff.

File: libs/components/lookup/src/lib/modules/search/search.component.ts [hunk 2]
```diff
@@ -207,7 +207,7 @@ export class SkySearchComponent implements OnDestroy, OnInit, OnChanges {
     if (searchText !== this.searchText) {
       this.searchText = searchText;
     }
-    this.clearButtonShown = searchText && searchText !== '';
+    this.clearButtonShown = !!(searchText && searchText !== '');
```
Comment (Blackbaud-SteveBrush) [inline libs/components/lookup/src/lib/modules/search/search.component.ts:210]: ☝️ Search and country field will be done in a future PR.

File: libs/components/indicators/src/lib/modules/tokens/tokens.component.ts [hunk 3]
```diff
@@ -207,7 +207,7 @@ export class SkyTokensComponent implements OnDestroy {
   private _disabled: boolean;
```
Comment (Blackbaud-PaulCrowder) [inline libs/components/indicators/src/lib/modules/tokens/tokens.component.ts:207]: Should we convert these to private-private in this PR?

File: libs/components/lookup/src/lib/modules/lookup/types/lookup-show-more-custom-picker-context.ts [hunk 4]
```diff
@@ -6,13 +6,13 @@ export class SkyLookupShowMoreCustomPickerContext {
   /**
    * Specifies the options that users can select in the custom picker.
    */
-  public items: any[];
+  public items!: any[];
```
Comment (Blackbaud-PaulCrowder) [inline libs/components/lookup/src/lib/modules/lookup/types/lookup-show-more-custom-picker-context.ts:9]: What's the purpose of `!` here?

Comment (Blackbaud-SteveBrush) [reply libs/components/lookup/src/lib/modules/lookup/types/lookup-show-more-custom-picker-context.ts:9]: The `items` property is always set when we create the context instance, so it won't ever be undefined. Better way to handle?

File: libs/components/indicators/src/lib/modules/tokens/tokens.component.ts (hunk 3 above)
Comment (Blackbaud-SteveBrush) [reply libs/components/indicators/src/lib/modules/tokens/tokens.component.ts:207]: I was going to tackle `@skyux/indicators` in a separate PR. I had to make these changes to get Lookup working.

**Review by Blackbaud-PaulCrowder — APPROVED** [review]

**Pull request merged**